src/scripts/main.js

```diff
@@ -1,3 +1,216 @@
 'use strict';
 
-// write code here
+// Sorting by header of column:
+
+const body = document.querySelector('body');
+const table = document.querySelector('table');
+const headers = table.querySelectorAll('th');
+const tBody = table.querySelector('tbody');
+const cells = tBody.querySelectorAll('td');
+let rows = [...tBody.querySelectorAll('tr')];
+const sortOrder = Array(headers.length).fill(true);
+
+function updateRows() {
+  rows = [...tBody.querySelectorAll('tr')];
+}
+
+headers.forEach((header, index) => {
+  header.addEventListener('click', () => {
+    setSortTable(index);
+  });
+});
+
+const setSortTable = (columnIndex) => {
+  updateRows();
+
+  const sortedRows = rows.sort((a, b) => {
+    const aValue = a.cells[columnIndex].textContent.trim();
+    const bValue = b.cells[columnIndex].textContent.trim();
+
+    if (!isNaN(aValue) && !isNaN(aValue)) {
```

Comment:
The condition `if (!isNaN(aValue) && !isNaN(aValue))` is incorrect as it checks `aValue` twice. It should be `if (!isNaN(aValue) && !isNaN(bValue))` to correctly check both values.